src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddDebugLogProcessorPass.php

```diff
@@ -22,22 +23,41 @@ public function process(ContainerBuilder $container)
         if (!$container->hasDefinition('profiler')) {
             return;
         }
-        if (!$container->hasDefinition('monolog.logger_prototype')) {
+
+        if ($container->hasDefinition('monolog.logger_prototype') && $container->hasDefinition('debug.log_processor')) {
+            $container->getDefinition('monolog.logger_prototype')
+                ->setConfigurator([__CLASS__, 'configureMonologLogger'])
+                ->addMethodCall('pushProcessor', [new Reference('debug.log_processor')])
+            ;
+
             return;
         }
-        if (!$container->hasDefinition('debug.log_processor')) {
+
+        if (!$container->hasDefinition('logger')) {
             return;
         }
 
-        $definition = $container->getDefinition('monolog.logger_prototype');
-        $definition->setConfigurator([__CLASS__, 'configureLogger']);
-        $definition->addMethodCall('pushProcessor', [new Reference('debug.log_processor')]);
+        $loggerDefinition = $container->getDefinition('logger');
+
+        if (Logger::class === $loggerDefinition->getClass()) {
+            $loggerDefinition
+                ->addMethodCall('enableDebug', [true])
```

Comment:
isn't this line redundant with the configurator?

Reply:
Hm indeed you’re right :thinking: